### PR TITLE
Add "regexCheck" for Facebook and Flipboard

### PR DESCRIPTION
**Pull request by theodosisathanasakis**
Facebook from: https://www.facebook.com/help/409473442437047 but I don't know how to handle this requirement: "They can't contain generic terms or extensions (.com, .net)".
Flipboard from: https://flipboard.helpshift.com/a/flipboard/?s=managing-your-account&f=help-with-username-issues&p=web

### CODE REVIEW

**Comment by hoadlck** [comment]
I really like https://regex101.com/ for trying out regular expressions.  Make sure to select the Python dialect before doing your experiments.

It appears that a Facebook user name must be 5 to 50 characters.  I am not clear if it is allowed to start with a number or not (so I assumed that it was ok).  And, I specifically do not allow .com, .org, and .net as a suffix.  So, that gives me the following regular expression.

> ^[a-zA-Z0-9](?:[a-zA-Z0-9\.]|-(?=[a-zA-Z0-9\.])){4,49}(?<!.com|.org|.net)$

I would check it out a bit before committing it.  I did basic checking, but these complicated regular expressions are tricky. :)

**Comment by theodosisathanasakis** [comment]
Yes, they can be very hard to undestand and check them but it is harder to find the requirements themselves, if any. 😂
I would try to find more and commit them.

**Comment by TheYahya** [comment]
And there's exceptions too, like facebook.com/zuck, It's only 4 character.

**Comment by theodosisathanasakis** [comment]
So, the official page of Facebook has wrong information? Unbelievable... With this in mind, there is no way that we know for sure if the regex field is right, for every website.

**Comment by theodosisathanasakis** [comment]
Should I do any other commits in order to close / merge this PR?

**Comment by TheYahya** [comment]
Length of 5 wouldn't be true for facebook, since the facebook.com/zuck username is exists.

File: data.json [hunk 1]
```diff
@@ -14,7 +14,8 @@
   "Facebook": {
     "url": "https://www.facebook.com/{}",
     "urlMain": "https://www.facebook.com/",
-    "errorType": "status_code"
+    "errorType": "status_code",
+    "regexCheck": "^([a-zA-Z0-9.]){5,}$"
```
Comment (TheYahya) [inline data.json]: Length of 5 wouldn't be true for facebook, since the `facebook.com/zuck` username is exists.

Comment (theodosisathanasakis) [reply data.json]: So, what do I need to do?
Should I delete the regex for Facebook?
Modify it as mr. hoadlck suggested _"^[a-zA-Z0-9](?:[a-zA-Z0-9\.]|-(?=[a-zA-Z0-9\.])){4,49}(?<!.com|.org|.net)$"_.
Or modify it in order for it to allow sequencies <50 letters and not contain .com .org .net?

**Comment by TheYahya** [comment]
You add `install_packages.sh` to your commit, Thats because you didn't separate your PRs with branch. Actually you should create a new brach for each PR/Patch... you going to contribute to so the PRs didn't mess up.
There's a great principle that popular principle that most open source projects on github kind of follow that, You can take a look at it: [git flow](https://nvie.com/posts/a-successful-git-branching-model/)
And if you had any question, Me and @nareddyt will walk you through.

**Comment by nareddyt** [comment]
It seems like you pushed `master` when you were making changes in PR #72. I'm not sure how you managed to get those changes in this branch.

You just need to checkout `master` on your computer and delete the [last commit](https://github.com/TheYahya/sherlock/pull/64/commits/61e82cd0b8f09a11c63153c14d2fe41a2627193e). Since this is the latest commit on the branch, you can just use `git reset`. Then force push to master.

On your fork:

```
git status (make sure you are on master, and there is no other work to commit)
git checkout master
git reset HEAD~1
git log (make sure your commits look correct)
git push -f
```

Once you do this, you can make another commit with the requested changes.

**Comment by nareddyt** [comment]
Ignore my last comment, it seems like you pushed another commit to your `master` branch somehow. You basically need to do what you did in #72, but this time for master branch. You need to "delete" [this commit](https://github.com/TheYahya/sherlock/pull/64/commits/f84629cc460f1501f49dceef6fbffbc40106981c).

**Comment by theodosisathanasakis** [comment]
I committed the right files, but there is an error in pushing them.
The error is:

fatal: The current branch master has no upstream branch.
To push the current branch and set the remote as upstream, use

    git push --set-upstream origin master

Although, I tried the suggested command, I got this:

To https://github.com/theodosisathanasakis/sherlock
 ! [rejected]        master -> master (non-fast-forward)
error: failed to push some refs to 'https://github.com/theodosisathanasakis/sherlock'
hint: Updates were rejected because the tip of your current branch is behind
hint: its remote counterpart. Integrate the remote changes (e.g.
hint: 'git pull ...') before pushing again.
hint: See the 'Note about fast-forwards' in 'git push --help' for details.

Any suggestions?

**Comment by nareddyt** [comment]
Yes, the first time you push, you need to tell git which branch on GitHub to push to. That's what `--set-upstream` does.

However, you still need to force push with `--force`. This is required because you deleted commits. If you're just adding commits, no need for `--force`.

So combine them both!

`git push --force --set-upstream origin master`

**Pull request merged**